Implementing support for custom sample apps, locally and via remote git repository

m-lukas opened the pull request:
- Replace sub-command structure (php, golang, all ...) with args
-> add support for specifying multiple apps at one
- Implement url lookup, use of git functionality, local download
- Refactor ExitGonut
- Fix hostname bug in sample app pinging
- Add test cases for localization functions
- Add `all` option to `-s` flag

Fix #22 and #24

m-lukas committed:
Implement pushing of custom, local sample apps

This commit implements the first step for the testing of custom
sample apps. It adds a new 'push custom' command which can be used
to push local sample apps in the fomat: 'file:/<path>' or 'file:///<path>'.

m-lukas committed:
Add url regex link and comment

m-lukas committed:
Refactor push command and add multi-app push

This comment refactors the push command to treat the sample app names
as args instead of sub-commands. This enables also the selectiv push
of multiple apps.

m-lukas committed:
Add git clone functionality for remote sample app urls

Implements the downloading (clone/pull) of sample apps from urls.

m-lukas committed:
Fix dynamic cf app hostname bug

There was an bug that gonut was using the appName as hostname but for
occupied urls and in other cases, cf creates a dynamic hostname.
This was fixed by retrieving and using the hostname.

m-lukas committed:
Refactor code and small fixes

This commit moves the remote samle app downloading and loading into
the internal package, removes the regex url check and restructures ExitGonut()

m-lukas committed:
Add test cases for location.go

This commit adds test cases for cloning, pulling and the integration
with the sample app loading via a local or public URL.

m-lukas committed:
Add 'all' option to the '-s' (stack) flag

This commit adds a new 'all' option to push sample apps in a row with
every installed stack. Furthermore, it refactors the stack and buildpack
check.

HeavyWombat committed:
Merge branch 'develop' into feature/custom-apps

HeavyWombat commented on internal/gonut/cmd/push.go:
```diff
@@ -123,6 +126,7 @@ var sampleApps = []sampleApp{
 		caption:       ".NET",
 		command:       "dotnet",
 		buildpack:     "dotnet-core",
+		aliases:       []string{"node"},
```
Is that a copy/paste error? That should not be an alias for that language.

HeavyWombat commented on internal/gonut/cmd/push.go:
```diff
@@ -131,6 +135,7 @@ var sampleApps = []sampleApp{
 		caption:       "Binary",
 		command:       "binary",
 		buildpack:     "binary_buildpack",
+		aliases:       []string{"node"},
```
Is that a copy/paste error? That should not be an alias for that language.

HeavyWombat commented on internal/gonut/cmd/push.go:
```diff
@@ -139,16 +144,21 @@ var sampleApps = []sampleApp{
 		caption:       "Java",
 		command:       "java",
 		buildpack:     "java_buildpack",
+		aliases:       []string{"node"},
```
Is that a copy/paste error? That should not be an alias for that language.

HeavyWombat reviewed (changes requested):
Thanks for the PR. Could you have a look at the additional aliases that seem wrong?

m-lukas committed:
Remove/Update fault sample app aliases

m-lukas committed:
Merge branch 'feature/custom-apps' of https://github.com/m-lukas/gonut into feature/custom-apps

m-lukas commented:
Fixed :)

m-lukas committed:
Add alias check to push command

Before there was no implementation of using aliases as command
replacements. This commit adds a name check which enables this
feature.
